package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -458,74 +463,164 @@ class NCDFReader(base.ReaderBase):
              'length': 'Angstrom',
              'velocity': 'Angstrom/ps',
              'force': 'kcal/(mol*Angstrom)'}
+
     _Timestep = Timestep
 
     def __init__(self, filename, n_atoms=None, mmap=None, **kwargs):
+
+        def verify_units(eval_unit, expected_units):
+            if eval_unit.decode('utf-8') not in expected_units:
+                errmsg = ("NETCDFReader currently assumes that the trajectory "
+                          "was written in units of {0} instead of {1}".format(
+                           expected_units, eval_unit.decode('utf-8')))
+                raise NotImplementedError(errmsg)
+
         self._mmap = mmap
 
         super(NCDFReader, self).__init__(filename, **kwargs)
 
         self.trjfile = scipy.io.netcdf.netcdf_file(self.filename,
                                                    mmap=self._mmap)
 
-        if not ('AMBER' in self.trjfile.Conventions.decode('utf-8').split(',') or
-                'AMBER' in self.trjfile.Conventions.decode('utf-8').split()):
+        # AMBER NetCDF files should always have a convention
+        try:
+            conventions = self.trjfile.Conventions
+            if not ('AMBER' in conventions.decode('utf-8').split(',') or
+                    'AMBER' in conventions.decode('utf-8').split()):
+                errmsg = ("NCDF trajectory {0} does not conform to AMBER "
+                          "specifications, "
+                          "http://ambermd.org/netcdf/nctraj.xhtml "
+                          "('AMBER' must be one of the token in attribute "
+                          "Conventions)".format(self.filename))
+                logger.fatal(errmsg)
+                raise TypeError(errmsg)
+        except AttributeError:
+            errmsg = "NCDF trajectory {0} is missing Conventions".format(
+                      self.filename)
+            logger.fatal(errmsg)
+            raise AttributeError(errmsg)
```

Comment:
This should be either a `ValueError` (data contains wrong/missing information) or [MDAnalysis.NoDataError](https://www.mdanalysis.org/mdanalysis/documentation_pages/exceptions.html#MDAnalysis.exceptions.NoDataError) – opinions on which one? What do we use elsewhere?

Reply:
So looking through all the other coordinate readers/writers, it looks like `NoDataError` is exclusively used by writers. `ValueError` is more commonly used for this, so as per your suggestion I'll swap all the `AttributeError` and `KeyError` instances to that.

Reply:
Should be dealt with in 54da919.